README.md

```diff
@@ -7,29 +7,310 @@
+```
+Return a list of Peripherals that have been scanned and match any of the supplied identifiers. Returns empty list if an empty list is passed.
+```dart
+  Future<List<Peripheral>> connectedDevices(List<String> serviceUUIDs);
+```
+Returns a list of connecterd Peripherals that have at least one service with UUID matching any of the supplied UUIDs. Returns empty list if an empty list is passed.
+#### Cancelling asynchronous operations
+```dart
+  Future<void> cancelTransaction(String transactionId);
+```
+Some of the asynchronous operations done by the library do not have a complementary `stop...()` function, but accept an optional parameter `transactionId`, eg. reading from characteristic. This function allows the user to break the execution of such operation, which will immediately return with BleError and errorCode `operationCancelled`.
 
-    Copyright 2018 Polidea Sp. z o.o
+**IMPORTANT**
+The cancelled operation will still be completed (perhaps with error), but the user will not be notified about its result.
+### Peripheral
+Object representing a peripheral. Allows for managing connection, discovery and serves as a shortcut to characteristic operations, if the user knows the UUIDs of both service and characteristic.
+#### Connection
+```dart
+  Future<void> connect(
+          {bool isAutoConnect = false,
+          int requestMtu,
+          bool refreshGatt = false,
+          Duration timeout});
+```
+Attemps to connect to the peripheral.
+`requestMtu` defaults to 23.
+`refreshGatt` forces GATT to refresh its cache; Android-specific.
+If connection has not been established by `timeout`, the operation fails. `timeout` defaults to 30 seconds.
+```dart
+  Stream<PeripheralConnectionState> observeConnectionState(
+          {bool emitCurrentValue = false,
+          bool completeOnDisconnect = false});
+```
+Return a stream containing changes to the connection state. By default doesn't emit current state, nor terminates the stream after disconnecting.
+**Note:** due ambiguities concerning `disconnecting` state, current implementation never emits `disconnecting`, only `connecting`, `connected`, `disconnected`.
 
-    Licensed under the Apache License, Version 2.0 (the "License");
-    you may not use this file except in compliance with the License.
-    You may obtain a copy of the License at
+```dart
+  Future<bool> isConnected();
+```
+Returns true if the peripheral is currently connected.
 
-       http://www.apache.org/licenses/LICENSE-2.0
+```dart
+  Future<void> disconnectOrCancelConnection();
+```
+Terminates connection or any attempt to connect.
+#### Obtaining services and characteristics
+```dart
+  Future<void> discoverAllServicesAndCharacteristics({String transactionId}) =>
+      _manager.discoverAllServicesAndCharacteristics(this, transactionId);
+```
+Runs the discovery process, caching all discovered services and characteristics in the native parts.
+**Must be run before `services()` or `characteristics()`**
```

Comment:
Must be run before services(), characteristics() and also before every attempt to perform IO operations on characteristics (eg. from peripheral level)